llvm/lib/Transforms/Utils/FunctionComparator.cpp

```diff
@@ -836,6 +837,54 @@ int FunctionComparator::cmpValues(const Value *L, const Value *R) const {
   return cmpNumbers(LeftSN.first->second, RightSN.first->second);
 }
 
+int FunctionComparator::cmpDistinctMetadata(const Function *L,
+                                            const Function *R) {
+  // Iterate over the basic blocks and instructions in Function L
+  for (const BasicBlock &BBL : *L) {
+    for (const Instruction &InstL : BBL) {
+      for (unsigned i = 0, e = InstL.getNumOperands(); i != e; ++i) {
+        if (MetadataAsValue *MDL =
+                dyn_cast<MetadataAsValue>(InstL.getOperand(i))) {
```

Comment:
I think the early return pattern should be used to reduce nesting:
```
MetadataAsValue *MDL = ...;
if (!MDL)
  continue;
```
There are many other instances of this below.